uppsrc/plugin/ppm/ppm.cpp

```diff
@@ -16,28 +16,59 @@ bool PPMRaster::Create()
 		SetError();
 		return false;
 	}
-
-	try {
-		if(stream.GetLine() != "P6")
-			return false;
-		String h = stream.GetLine();
-		CParser p(h);
-		size.cx = p.ReadInt();
-		size.cy = p.ReadInt();
-		if(size.cx <= 0 || size.cx > 99999 || size.cy <= 0 || size.cy >= 99999)
-			return false;
-		h = stream.GetLine();
-		CParser p1(h);
-		int maxval = p1.ReadInt();
-		if(maxval <= 0 || maxval > 65535)
-			return false;
-		is16 = maxval > 255;
-		pixel_pos = stream.GetPos();
-		return true;
+	
+	auto SkipWhitespaces = [&] {
+		for(;;) {
+			while(IsSpace(stream.Peek()))
+				(void) stream.Get();
+			if(stream.Peek() == '#')
+				(void) stream.GetLine();
+			else
+				break;
+		}
+	};
+	
+	auto ReadInt = [&] {
+		int n = 0, c;
+		while((c = stream.Get()) >= '0' && c <= '9') {
+			// Overflow check: ensure n * 10 + digit doesn't overflow
+			if(n > (INT_MAX - 9) / 10) {
+				SetError();
+				return -1;
+			}
+			n = n * 10 + (c - '0');
+		}
+		return n;
+	};
+	
+	// PPM P6 format
+	if(stream.Get() != 'P' || stream.Get() != '6') {
+		SetError();
+		return false;
+	}
+	
+	SkipWhitespaces();
+	if(size.cx = ReadInt(); size.cx <= 0 || size.cx > 99999) {

```

Comment:
What is the origin of 99999? Consider defining it as a constant. This limit has been used in two places.

Reply:
Apparenty, it is arbitrary. I removed them and used `std::numeric_limits<word>::max()` for all.